pilot/pkg/bootstrap/server.go

```diff
@@ -214,7 +216,54 @@ func NewServer(args *PilotArgs) (*Server, error) {
 
 	}
 
-	// initDNSListener() must be called after the createCA()
+	// Setup the root cert chain and caBundlePath - before calling initDNSListener.
+	if features.PilotCertProvider.Get() == KubernetesCAProvider {
```

Comment:
nit: why not just have this in initDnsListener? Right now this function is mostly just calling out to a bunch of other functions so it seems out of place to have all of this logic here

Reply:
I don't remember why I put it here. `initDnsListener()`sounds like the right place to put it.